evm_loader/program/src/account_storage/mod.rs

```diff
@@ -145,11 +145,11 @@ pub trait AccountStorage {
 #[must_use]
 pub fn generate_fake_block_hash(slot: Slot) -> [u8; 32] {
     let slot_bytes: [u8; 8] = slot.to_be_bytes();
-    let non_null_bytes: Vec<_> = slot_bytes.into_iter().skip_while(|&n| n == 0).collect();
-    let non_null_len = non_null_bytes.len();
+    let slot_vec: Vec<_> = std::iter::once(0)

```

Comment:
I think, it also could be cheaper not to create temporary vector, but just use in `copy_from_slice` sub-slice from `slot_bytes`, started from non-zero.